lib/windows/plugin/containers/FirmwareDialogContainer.js

```diff
@@ -52,7 +53,10 @@ function mapStateToProps(state) {
 function mapDispatchToProps(dispatch) {
     return {
         onConfirmUpdateFirmware: port => dispatch(FirmwareDialogActions.updateFirmware(port)),
-        onCancel: () => dispatch(FirmwareDialogActions.hideDialog()),
+        onCancel: port => {
+            dispatch(FirmwareDialogActions.hideDialog());
+            dispatch(SerialPortActions.selectPort(port));
```

Comment:
Now that we are selecting the port even when the user clicks Cancel, we may want to adjust the dialog text. Currently it says: "The development kit will be programmed with the required firmware. Would you like to proceed?".

How about "Would you like to program the development kit with the required firmware?", and then change the buttons to say "Yes" and "No, proceed without programming". This would require a small change in `ConfirmationDialog` to allow `okButtonText` and `cancelButtonText` to be passed in as props.